### ISSUE TITLE
TestThreadingLayerPriority tests fail on openSUSE rpm build

### ISSUE THREAD

**Issue by bnavigator** [numba/numba#7746]
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as ~'python bug.py'~.
  `python3.9 -m numba.runtests -v -b --exclude-tags=long_running -m 12 -- numba.tests`

## Description
Hi, I am trying to package numba 0.55.0 for openSUSE Tumbleweed, which is required so that we can get the Python 3.10 stack into a sane state.

## Test log

I encounter these test failures, (it does not matter if I try with Python 3.8, 3.9 or 3.10)

```
[    9s] Executing(%check): /bin/sh -e /var/tmp/rpm-tmp.dP3Cuj
[    9s] + umask 022
[    9s] + cd /home/abuild/rpmbuild/BUILD
[    9s] + cd numba-0.55.0
[    9s] + mv numba numba_temp
[    9s] + export NUMBA_PARALLEL_DIAGNOSTICS=1
[    9s] + NUMBA_PARALLEL_DIAGNOSTICS=1
[    9s] ++ '[' -f _current_flavor ']'
[    9s] ++ true
[    9s] + last_flavor=
[    9s] + '[' -z '' ']'
[    9s] + last_flavor=tmp
[    9s] + '[' tmp '!=' python39 ']'
[    9s] + '[' -d build ']'
[    9s] + '[' -d _build.python39 ']'
[    9s] + echo python39
[    9s] + python_flavor=python39
[    9s] + /usr/bin/numba-3.9 -s
[   10s] System info:
[   10s] --------------------------------------------------------------------------------
[   10s] __Time Stamp__
[   10s] Report started (local time)                   : 2022-01-14 17:14:42.069699
[   10s] UTC start time                                : 2022-01-14 17:14:42.069701
[   10s] Running time (s)                              : 0.256123
[   10s] 
[   10s] __Hardware Information__
[   10s] Machine                                       : x86_64
[   10s] CPU Name                                      : znver2
[   10s] CPU Count                                     : 12
[   10s] Number of accessible CPUs                     : 12
[   10s] List of accessible CPUs cores                 : 0 1 2 3 4 5 6 7 8 9 10 11
[   10s] CFS Restrictions (CPUs worth of runtime)      : None
[   10s] 
[   10s] CPU Features                                  : 64bit adx aes avx avx2 bmi bmi2
[   10s]                                                 clflushopt clwb clzero cmov cx16
[   10s]                                                 cx8 f16c fma fsgsbase fxsr lzcnt
[   10s]                                                 mmx movbe mwaitx pclmul popcnt
[   10s]                                                 prfchw rdpid rdrnd rdseed sahf sha
[   10s]                                                 sse sse2 sse3 sse4.1 sse4.2 sse4a
[   10s]                                                 ssse3 wbnoinvd xsave xsavec
[   10s]                                                 xsaveopt xsaves
[   10s] 
[   10s] Memory Total (MB)                             : 32005
[   10s] Memory Available (MB)                         : 27898
[   10s] 
[   10s] __OS Information__
[   10s] Platform Name                                 : Linux-5.15.12-1-default-x86_64-with-glibc2.34
[   10s] Platform Release                              : 5.15.12-1-default
[   10s] OS Name                                       : Linux
[   10s] OS Version                                    : #1 SMP Wed Dec 29 14:50:16 UTC 2021 (375fcb8)
[   10s] OS Specific Version                           : ?
[   10s] Libc Version                                  : glibc 2.34
[   10s] 
[   10s] __Python Information__
[   10s] Python Compiler                               : GCC
[   10s] Python Implementation                         : CPython
[   10s] Python Version                                : 3.9.9
[   10s] Python Locale                                 : en_US.UTF-8
[   10s] 
[   10s] __Numba Toolchain Versions__
[   10s] Numba Version                                 : 0.55.0
[   10s] llvmlite Version                              : 0.38.0
[   10s] 
[   10s] __LLVM Information__
[   10s] LLVM Version                                  : 11.0.1
[   10s] 
[   10s] __CUDA Information__
[   10s] CUDA Device Initialized                       : False
[   10s] CUDA Driver Version                           : ?
[   10s] CUDA Runtime Version                          : ?
[   10s] CUDA NVIDIA Bindings Available                : ?
[   10s] CUDA NVIDIA Bindings In Use                   : ?
[   10s] CUDA Detect Output:
[   10s] None
[   10s] CUDA Libraries Test Output:
[   10s] None
[   10s] 
[   10s] __SVML Information__
[   10s] SVML State, config.USING_SVML                 : False
[   10s] SVML Library Loaded                           : False
[   10s] llvmlite Using SVML Patched LLVM              : False
[   10s] SVML Operational                              : False
[   10s] 
[   10s] __Threading Layer Information__
[   10s] TBB Threading Layer Available                 : True
[   10s] +-->TBB imported successfully.
[   10s] OpenMP Threading Layer Available              : True
[   10s] +-->Vendor: GNU
[   10s] Workqueue Threading Layer Available           : True
[   10s] +-->Workqueue imported successfully.
[   10s] 
[   10s] __Numba Environment Variable Information__
[   10s] NUMBA_PARALLEL_DIAGNOSTICS                    : 1
[   10s] 
[   10s] __Conda Information__
[   10s] Conda not available.
[   10s] 
[   10s] __Installed Packages__
[   10s] Package            Version
[   10s] ------------------ --------
[   10s] apipkg             2.1.0
[   10s] appdirs            1.4.4
[   10s] attrs              21.4.0
[   10s] backcall           0.2.0
[   10s] cffi               1.15.0
[   10s] decorator          5.1.1
[   10s] importlib-metadata 4.8.2
[   10s] iniconfig          0.0.0
[   10s] ipython            7.30.1
[   10s] jedi               0.18.1
[   10s] Jinja2             3.0.3
[   10s] llvmlite           0.38.0
[   10s] MarkupSafe         2.0.1
[   10s] matplotlib-inline  0.1.3
[   10s] more-itertools     8.10.0
[   10s] numba              0.55.0
[   10s] numpy              1.21.4
[   10s] ordered-set        4.0.2
[   10s] packaging          21.3
[   10s] parso              0.8.3
[   10s] pexpect            4.8.0
[   10s] pickleshare        0.7.5
[   10s] pip                20.2.4
[   10s] pluggy             1.0.0
[   10s] prompt-toolkit     3.0.8
[   10s] psutil             5.9.0
[   10s] ptyprocess         0.7.0
[   10s] py                 1.10.0
[   10s] pybind11           2.8.1
[   10s] pycparser          2.21
[   10s] Pygments           2.9.0
[   10s] pyparsing          3.0.6
[   10s] pytest             6.2.5
[   10s] PyYAML             6.0
[   10s] scipy              1.6.3
[   10s] setuptools         57.4.0
[   10s] TBB                2021.4.0
[   10s] toml               0.10.2
[   10s] traitlets          5.1.1
[   10s] wcwidth            0.2.5
[   10s] zipp               3.7.0
[   10s] 
[   10s] No errors reported.
[   10s] 
[   10s] 
[   10s] __Warning log__
[   10s] Warning (cuda): CUDA driver library cannot be found or no CUDA enabled devices are present.
[   10s] Exception class: <class 'numba.cuda.cudadrv.error.CudaSupportError'>
[   10s] Warning: Conda not available.
[   10s]  Error was [Errno 2] No such file or directory: 'conda'
[   10s] 
[   10s] Warning (no file): /sys/fs/cgroup/cpuacct/cpu.cfs_quota_us
[   10s] Warning (no file): /sys/fs/cgroup/cpuacct/cpu.cfs_period_us
[   10s] --------------------------------------------------------------------------------
[   10s] If requested, please copy and paste the information between
[   10s] the dashed (----) lines, or from a given specific section as
[   10s] appropriate.
[   10s] 
[   10s] =============================================================
[   10s] IMPORTANT: Please ensure that you are happy with sharing the
[   10s] contents of the information present, any information that you
[   10s] wish to keep private you should remove before sharing.
[   10s] =============================================================
[   10s] 
[   10s] + python3.9 -m numba.runtests -v -b --exclude-tags=long_running -m 12 -- numba.tests
[   11s] skipped CUDA tests
[   11s] skipped CUDA tests
[   11s] Parallel: 8758. Serial: 646
...
[  970s] ======================================================================
[  970s] FAIL: test_omp (numba.tests.test_parallel_backend.TestThreadingLayerPriority)
[  970s] ----------------------------------------------------------------------
[  970s] Traceback (most recent call last):
[  970s]   File "/usr/lib64/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 583, in test_omp
[  970s]     self.assertEqual(threading_layer_priority.strip(), env_var)
[  970s] AssertionError: '=========================================[528 chars]ueue' != 'omp tbb workqueue'
[  970s] - ================================================================================
[  970s] - ======== Parallel Accelerator Optimizing:  Function plus, <string> (6)  ========
[  970s] - ================================================================================
[  970s] - No source available
[  970s] - ------------------------------ After Optimisation ------------------------------
[  970s] - Parallel structure is already optimal.
[  970s] - --------------------------------------------------------------------------------
[  970s] - --------------------------------------------------------------------------------
[  970s] -  
[  970s]   omp tbb workqueue
[  970s] 
[  970s] ======================================================================
[  970s] FAIL: test_tbb (numba.tests.test_parallel_backend.TestThreadingLayerPriority)
[  970s] ----------------------------------------------------------------------
[  970s] Traceback (most recent call last):
[  970s]   File "/usr/lib64/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 590, in test_tbb
[  970s]     self.assertEqual(threading_layer_priority.strip(), env_var)
[  970s] AssertionError: '=========================================[528 chars]ueue' != 'tbb omp workqueue'
[  970s] - ================================================================================
[  970s] - ======== Parallel Accelerator Optimizing:  Function plus, <string> (6)  ========
[  970s] - ================================================================================
[  970s] - No source available
[  970s] - ------------------------------ After Optimisation ------------------------------
[  970s] - Parallel structure is already optimal.
[  970s] - --------------------------------------------------------------------------------
[  970s] - --------------------------------------------------------------------------------
[  970s] -  
[  970s]   tbb omp workqueue
[  970s] 
[  970s] ======================================================================
[  970s] FAIL: test_valid_env_var (numba.tests.test_parallel_backend.TestThreadingLayerPriority)
[  970s] ----------------------------------------------------------------------
[  970s] Traceback (most recent call last):
[  970s]   File "/usr/lib64/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 567, in test_valid_env_var
[  970s]     self.assertEqual(threading_layer_priority.strip(), env_var)
[  970s] AssertionError: '=========================================[528 chars]ueue' != 'tbb omp workqueue'
[  970s] - ================================================================================
[  970s] - ======== Parallel Accelerator Optimizing:  Function plus, <string> (6)  ========
[  970s] - ================================================================================
[  970s] - No source available
[  970s] - ------------------------------ After Optimisation ------------------------------
[  970s] - Parallel structure is already optimal.
[  970s] - --------------------------------------------------------------------------------
[  970s] - --------------------------------------------------------------------------------
[  970s] -  
[  970s]   tbb omp workqueue
[  970s] 
[  970s] ======================================================================
[  970s] FAIL: test_workqueue (numba.tests.test_parallel_backend.TestThreadingLayerPriority)
[  970s] ----------------------------------------------------------------------
[  970s] Traceback (most recent call last):
[  970s]   File "/usr/lib64/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 596, in test_workqueue
[  970s]     self.assertEqual(threading_layer_priority.strip(), env_var)
[  970s] AssertionError: '=========================================[528 chars] omp' != 'workqueue tbb omp'
[  970s] - ================================================================================
[  970s] - ======== Parallel Accelerator Optimizing:  Function plus, <string> (6)  ========
[  970s] - ================================================================================
[  970s] - No source available
[  970s] - ------------------------------ After Optimisation ------------------------------
[  970s] - Parallel structure is already optimal.
[  970s] - --------------------------------------------------------------------------------
[  970s] - --------------------------------------------------------------------------------
[  970s] -  
[  970s]   workqueue tbb omp
[  970s] 
[  970s] ----------------------------------------------------------------------
[  970s] Ran 9404 tests in 959.458s
[  970s] 
[  970s] FAILED (failures=4, skipped=510, expected failures=13)
```

Any hints what could be wrong with the test environment?

**Comment by gmarkall** [comment]
> Any hints what could be wrong with the test environment?

> ```
> [   10s] __Numba Environment Variable Information__
> [   10s] NUMBA_PARALLEL_DIAGNOSTICS                    : 1
> ```

I think this environment variable being set might be responsible for the extra output that is confusing the test.

**Comment by bnavigator** [comment]
I see. Not sure why it was defined (some old debugging attempt maybe), but removing it makes all tests pass. Thanks!

**Comment by gmarkall** [comment]
You're welcome - glad all tests are now passing!

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/issues/7458 notes that this might happen... I'll go and fix it.

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/pull/7749 fixes the underlying issue from #7458.
